### ISSUE TITLE
[Wav2Vec2] Possible bug with batched input

### ISSUE THREAD

**Issue by patrickvonplaten** [pytorch/fairseq#3227]
## 🐛 Bug

When using the Wav2Vec2 model with batched input (meaning that both `source` and `padding_mask` are provided), the input is different for padded input than it would be for the same non-padded input. At the moment this accounts for all models:

- Wav2Vec 2.0 Base
- Wav2Vec 2.0 Large
- Wav2Vec 2.0 Large (LV-60)*
- Wav2Vec 2.0 Large (LV-60) + Self Training *

IMO, the Base & Large models cannot really be fixed because the convolutional layers apply group norm over the time dimension meaning that adding 0's as padding tokens necessarily changes the result. See the following code snippet.

On the other hand the "new architectures" Large Lv-60 + Large Lv-60 + Self Training can be fixed as shown in #3228.

### To Reproduce

Please run the following code snippet to reproduce the error:

```python
import fairseq
import torch

# get model
wav2vec_path = "data/wav2vec_small_960h.pt"
model, cfg, task = fairseq.checkpoint_utils.load_model_ensemble_and_task(
    [wav2vec_path], arg_overrides={"data": "./data"}
)
model = model[0]
model.eval()

# create single input
input_wav_0 = torch.randn((1, 2000))
input_wav_1 = torch.randn((1, 3000))

# create batched input
batch_input_wav = torch.zeros((2, 3000))
batch_input_wav[0, :input_wav_0.shape[-1]] = input_wav_0
batch_input_wav[1, :input_wav_1.shape[-1]] = input_wav_1

# create padding mask
padding_mask = torch.zeros((2, 3000), dtype=torch.bool)
padding_mask[0, input_wav_0.shape[-1]:] = True

# run batch & single
output = model(source=input_wav_0, padding_mask=None)["encoder_out"]
batch_output = model(source=batch_input_wav, padding_mask=padding_mask)["encoder_out"]

# is equal?
print("Is batched forward and simple forward equal?", torch.allclose(output[:,0], batch_output[:output.shape[0], 0], atol=1e-1))
```

**Note**: It is assumed that both `https://dl.fbaipublicfiles.com/fairseq/wav2vec/dict.ltr.txt` and `https://dl.fbaipublicfiles.com/fairseq/wav2vec/wav2vec_small_960h.pt` were downloaded and stored in the folder `data`.

Also see [this](https://colab.research.google.com/drive/1rvP8N5WmQ4zReoH4yOAXb-ZjBSfsUZ09?usp=sharing) notebook that can be simply run to reproduce the error.

### Expected behavior

Both the padded and identical non-padded input should be equal in my opinion -> it seems like this is however not possible due to the layer norm / group norm layers in the model.

### Environment

 - fairseq Version (e.g., 1.0 or master): master
 - PyTorch Version (e.g., 1.0): 1.7.0+cu101
 - OS (e.g., Linux): Linux
 - How you installed fairseq (`pip`, source): !pip install git+https://github.com/pytorch/fairseq
 - Build command you used (if compiling from source): just pip
 - Python version: 3.6.9
 - CUDA/cuDNN version: No CUDA
 - GPU models and configuration: NO GPU

**Comment by patrickvonplaten** [comment]
This actually seems to be linked to #3118
